lib/grplatt.gi

```diff
@@ -3654,6 +3654,141 @@ function(G)
+        return q;
+    elif series = "A" then # Alt
+        return q;
+    elif series = "L" then # PSL
+        d := m;
+        if (d = 4 and q = 2) then return 8; fi;
+        if d = 2 then
+            if q = 9 then return 6; fi;
+            if q in [5,7,11] then return q; fi;
+        fi;
+        return (q^d-1)/(q-1);
+    elif series = "2A" then # PSU
+        d := m + 1;
+        if d = 3 and q = 5 then return 50; fi;
+        if d = 3 then return q^3 + 1; fi;
+        if d = 4 then return (q+1)*(q^3 + 1); fi;
+        if d mod 2 = 0 and q = 2 then return (2^(d-1)*(2^d - 1))/3; fi;
+        return ((q^d - (-1)^d)*(q^(d-1) + (-1)^d))/(q^2-1);
+    elif series = "B" then # P\Omega or O
+        d := 2*m +1;
+        if q = 3 and m > 2 then return (3^m)*(3^m - 1)/2; fi;
+        if q > 4 and m > 2 then return (q^(2*m)-1)/(q-1); fi;
+        #Special case : B(2,3) ~ 2A(3,2) = PSU(4,2)
+        if q = 3 and m = 2 then return 27;fi;
+        # B(2,2) is not a simple group.
+        #Special case : B(2,q) ~ C(2,q) = PSp(4,q)
+        if m=2 then return (q^(2*m) -1)/(q-1);fi;
+        #Special case : B(m,2) ~ C(m,2) = PSp(2*m,2)
+        if q = 2 then return (2^(m-1))*(2^m -1); fi;
+    elif series = "2B" then # Sz or _2 B^2
+        if 2^(Log2Int(q)) = q and Log2Int(q) mod 2 = 1 then
+            return q^2 + 1;
+        fi;
+    elif series = "C" then # PSp
+        d := 2*m;
+        if d=4 and q=2 then return 6;fi;
+        if d=4 and q=3 then return 27;fi;
+        if m>2 and q=2 then return (2^(m-1))*(2^m -1);fi;
+        if m>1 and q>2 then return (q^d -1)/(q-1);fi;
+    elif series = "D" then # P\Omega ^+ or O+
+        d := 2*m;
+        if m > 3 then
+            if q < 4 then return q^(m-1)*(q^m -1)/(q-1);
+            else return (q^(m-1) + 1)*(q^m-1)/(q-1); fi;
+        fi;
+    elif series = "2D" then # P\Omega ^- or O-
+        d := 2*m;
+        if m > 3 then return (q^m+1)*(q^(m-1)-1)/(q-1);fi;
+    elif series = "3D" then # ^3 D_4
+        return (q^8 + q^4 + 1)*(q+1);
+    elif series = "E" then #E_n(q)
+        d := m;
+        if d = 6 then return (q^9 - 1)*(q^8 + q^4 + 1)/(q-1);
+        elif d = 7 then return (q^14 - 1)*(q^9 + 1)*(q^5 -1)/(q-1);
+        elif d = 8 then return (q^30 - 1)*(q^12 + 1)*(q^10 + 1)*(q^6 + 1)/(q-1);
+        fi;
+    elif series = "2E" then #2E(6,q)
+        return (q^12 -1)*(q^6 - q^3 + 1)^(q^4 +1)/(q-1);
+    elif series = "F" then #F(4,q)
+        return (q^12 -1)*(q^4 + 1)/(q-1);
+    elif series = "2F" then #2F(4,q)
+        #special case : 2F4(2) ~ Tits
+        if q = 2 then return 1600; fi;
+        return (q^6 + 1)*(q^3 + 1)*(q+1);
+    elif series = "G" then #G(2,q)
+        if q = 3 then return 351; fi;
+        if q = 4 then return 416; fi;
+        return (q^6 -1)/(q-1);
+    elif series = "2G" then #2G(2,q)
+        b := Int(Log(Float(q)) / Log(3.0));
+        if 3^(b+1) = q then b := b+1; fi; #just a safety net
+        if q = 3^b and b mod 2 = 1 then return q^3 + 1; fi;
```

Comment:
As above, shouldn't that condition always be satisfied here? If so, I'd instead turn it into an `Assert`, or into an error (`if cond not met then Error(...); fi;`)